dali/operators/ssd/box_encoder.cuh

```diff
@@ -77,6 +78,7 @@ class BoxEncoder<GPUBackend> : public Operator<GPUBackend> {
  private:
   static constexpr int kBoxesOutputDim = 2;
   static constexpr int kLabelsOutputDim = 1;
+  int curr_batch_size_;
```

Comment:
I think at some point you gave a rationale why not make the `curr_batch_size_` a member.